bibtex.go

```diff
@@ -115,6 +131,46 @@ func (entry *BibEntry) AddField(name string, value BibString) {
 	entry.Fields[strings.TrimSpace(name)] = value
 }
 
+//prettyStringAppend appends the pretty print string for BibEntry to buf using priority to order the keys
+func (entry *BibEntry) prettyStringAppend(buf *bytes.Buffer, priority map[string]int) {
+	fmt.Fprintf(buf, "@%s{%s,\n", entry.Type, entry.CiteName)
+
+	// Determine key order.
+	keys := []string{}
+	for key := range entry.Fields {
+		keys = append(keys, key)
+	}
+
+	sort.Slice(keys, func(i, j int) bool {
+		pi, pj := priority[keys[i]], priority[keys[j]]
+		return pi < pj || (pi == pj && keys[i] < keys[j])
+	})
+
+	// Write fields.
+	tw := tabwriter.NewWriter(buf, 1, 4, 1, ' ', 0)
+	for _, key := range keys {
+		value := entry.Fields[key].String()
+		format := stringformat(value)
+		fmt.Fprintf(tw, "    %s\t=\t"+format+",\n", key, value)
+	}
+	tw.Flush()
+	buf.WriteString("}\n")
+
+}
+
+//PrettyStringCustomOrder pretty prints BibEntry with fields ordered as they appear
+//in keyOrder
+func (entry *BibEntry) PrettyStringCustomOrder(keyOrder []string) string {
```

Comment:
I'd probably prefer [functional options](https://golang.cafe/blog/golang-functional-options-pattern.html) to pass to `PrettyString` to override the extra options, than adding a new method every time we need a new parameter